tests/test_landfill_sampler.py

```diff
@@ -0,0 +1,99 @@
+from copy import deepcopy
+
+import pytest
+from pyspark.sql import functions as F
+
+from mozetl.landfill import sampler
+
+
+@pytest.fixture
+def sample_document():
+    return {
+        'content': {"payload": {"foo": "bar"}},
+        'meta': {
+            u'Content-Length': u'7094',
+            u'Date': u'Sun, 19 Aug 2018 15:08:00 GMT',
+            u'Host': u'incoming.telemetry.mozilla.org',
+            'Hostname': u'ip-1.1.1.1',
+            'Timestamp': 1534691279765301222L,
+            'Type': u'telemetry-raw',
+            u'User-Agent': u'pingsender/1.0',
+            u'X-Forwarded-For': u'127.0.0.1',
+            u'X-PingSender-Version': u'1.0',
+            u'args': u'v=4',
+            u'protocol': u'HTTP/1.1',
+            u'remote_addr': u'1.1.1.1',
+            u'uri': u'/submit/telemetry/doc-id/main/Firefox/61.0.2/release/20180807170231'
+        }
+    }
+
+
+@pytest.fixture
+def generate_data(spark, sample_document):
+    """Update the meta-field using provided snippets."""
+    def _update_meta(snippet):
+        doc = deepcopy(sample_document)
+        doc["meta"].update(snippet)
+        return doc
+
+    def _generate_data(snippets):
+        return spark.sparkContext.parallelize(map(_update_meta, snippets))
+
+    return _generate_data
```

Comment:
I guess you could just use `lambda` here. Might be a little more concise.

Reply:
I'm going to leave it as-is, I think the named function helps a little bit in documenting what it's doing.